sa/_db-next/migrations/20180124141109_DropOrderStatusField.sql

```diff
@@ -0,0 +1,12 @@
+
+-- +goose Up
+-- SQL in section 'Up' is executed when this migration is applied
+
+ALTER TABLE `orders` DROP COLUMN `status`;
+ALTER TABLE `orders` ADD COLUMN `beganProcessing` BOOL NOT NULL DEFAULT FALSE;
```

Comment:
Note that deploying this PR before the migration is applied will result in errors due to the missing column. Let's do this: Start a separate PR that just adds the `beganProcessing` column, and let's land that in advance of everything else. Then Ops can apply the migration without needing to time it precisely with the staging deploy.

Once this new code is deployed, we can commit a separate migration to delete the no-longer-needed status field.

Reply:
Sorry, I checked the code after I wrote this and realized a problem: Right now, `GetOrder` uses `dbMap.Get(orderModel{}, *req.Id)`, which I *think* will just select all of the fields in the table and expect them to match the fields in the model. We've run into this problem before with other models, which led us to switch to using `Select(...)` with an explicit list of field names. If we were doing this in prod, I would want to deploy a change like that, and then the migration, and then the second change (or gate the new code behind a feature flag). However, given that this only affects staging, I'm willing to take a longer-than usual staging outage for the deploy, and do the migration as part of the deploy. So, this migration can stay as-is, but please file an Ops ticket to ensure that the deployer knows that migrations do have to be applied during the deploy, contrary to our usual practice.

Reply:
This migration is from https://github.com/letsencrypt/boulder/pull/3402 and not specific to this PR.

I'll make sure to file the ticket mentioned.

Reply:
Ticket is filed.